lib/marc_record.rb

```diff
@@ -0,0 +1,29 @@
+class MarcRecord
+  attr_reader :nypl_source, :id, :is_partner
+
+  def initialize(nypl_source, id)
+    @nypl_source = nypl_source
+    @id = id
+    @log_data = {}
+  end
+
+  def is_partner?
+    result = nypl_source == "recap-cul" || nypl_source == "recap-pul"
+    @log_data[:is_partner?] = result
+    return result
+  end
+
+  def get_platform_api_data
+    response = $platform_api.get(api_path)
+
+    raise NotFoundError unless response["data"]
+
+    data = response["data"]
+
+    @is_partner = is_partner?
```

Comment:
I'm not sure `@is_partner` is needed as just calling `is_partner?` seems equivalent and cheap. However, if you want to keep `@is_partner`, could it be established in `initialize`? Determining whether or not something is a partner record doesn't depend on any fetched data.

Reply:
It does not depend on any fetched data, except for validating that the record exists. Just refactored a bit to only need that value once.

Reply:
Ahh existence. good point.